code/datums/status_effects/debuffs/fire_stacks.dm

```diff
@@ -72,6 +82,13 @@
 
 /datum/status_effect/fire_handler/proc/on_reapplying_stacks(datum/status_effect/fire_handler/our_status)
 
+/**
+ * Recalculates owner's cached firestacks
+ */
+
+/datum/status_effect/fire_handler/proc/recalculate_firestacks()
+	owner.fire_stacks = max(0, stacks) * stack_modifier
```

Comment:
It might be best to make an internal setter for fire_stacks, to avoid someone forgetting about it